src/Microsoft.AspNetCore.Razor.Tools/Program.cs

```diff
@@ -16,17 +17,31 @@ public static int Main(string[] args)
             var cancel = new CancellationTokenSource();
             Console.CancelKeyPress += (sender, e) => { cancel.Cancel(); };
 
+            var outputWriter = new StringWriter();
+            var errorWriter = new StringWriter();
+
             // Prevent shadow copying.
             var loader = new DefaultExtensionAssemblyLoader(baseDirectory: null);
-            var checker = new DefaultExtensionDependencyChecker(loader, Console.Out, Console.Error);
+            var checker = new DefaultExtensionDependencyChecker(loader, outputWriter, errorWriter);
 
             var application = new Application(
                 cancel.Token,
                 loader,
                 checker,
-                (path, properties) => MetadataReference.CreateFromFile(path, properties));
+                (path, properties) => MetadataReference.CreateFromFile(path, properties),
+                outputWriter,
+                errorWriter);
+
+            var result = application.Execute(args);
+
+            var output = outputWriter.ToString();
+            var error = errorWriter.ToString();
+
+            // This will no-op if server logging is not enabled.
+            ServerLogger.Log(output);
+            ServerLogger.Log(error);
```

Comment:
We've already something [similar](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Razor.Tools/CompilerHost.cs#L53) for the code path when the server is already started. But I missed logging this code path which gets called during the server start.